autoqubo/utils.py

```diff
@@ -11,52 +11,54 @@ def energy(q, x, offset=0):
         :param offset:
         :return:
         """
-        return x@q@x + offset
+        return x @ q @ x + offset
 
     @staticmethod
     def get_matrix_dict_repr(q):
-       """
-       Returns dict representation of a QUBO matrix.
-       :param q:
-           QUBO matrix
-       :return:
-           the input matrix as a dict object
-       """
-       dict_repr = {}
-       for i, j in product(range(len(q)), repeat=2):
-           if q[i, j] != 0:
-               dict_repr[(i, j)] = q[i, j]
-       return dict_repr
+        """
+        Returns dict representation of a QUBO matrix.
+        :param q:
+            QUBO matrix
+        :return:
+            the input matrix as a dict object
+        """
+        dict_repr = {}
+        for i, j in product(range(len(q)), repeat=2):
+            if q[i, j] != 0:
+                dict_repr[(i, j)] = q[i, j]
+        return dict_repr
 
     @staticmethod
     def get_solution_vector_repr(d):
-       """
-       Returns vector representation of solution vector.
-       :param q:
-           solution vector
-       :return:
-           the input solution as a vector
-       """
-       return [d[i] for i in range(len(d))]
+        """
+        Returns vector representation of solution vector.
+        :param q:
+            solution vector
+        :return:
+            the input solution as a vector
+        """
+        return [d[i] for i in range(len(d))]
 
     @staticmethod
-    def solve(q, offset=0):
-       """
-       Returns solutions to a given QUBO problem.
-       :param q:
-           QUBO matrix
-       :param offset:
-           offset
-       :return: solutions, energies
-       """
-       from dwave_qbsolv import QBSolv
-       import warnings
+    def solve(q, offset=0, target=None, timeout=60):
+        """
+        Returns solutions to a given QUBO problem.
+        :param q:
+            QUBO matrix
+        :param offset:
+            offset
+         : param target:
+             target energy as stopping criterion
+        :return: solutions, energies
+        """
+        from dwave_qbsolv import QBSolv
```

Comment:
@gserban @lpodl  I think it would be nice to remove the QBSolv dependency, it's deprecated and it's Dwave.

Maybe I can ask Pieter to look for a non-deprecated solver and update this part. Thoughts?

Reply:
It's nice to have a QUBO solver at hand, but I don't know of any alternative to qbsolv...

Reply:
https://github.com/jtiosue/qubovert

This one seems a bit more neutral. But we will handle this on our side, as this is a bit heavy.